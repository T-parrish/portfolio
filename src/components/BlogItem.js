import React from 'react';


export const BlogItem = (props) => (
    <div className='blog__container'> 
        <h4 className='blog__title'> {props.content.title} </h4>
        <p className='blog__description'> {props.content.description} </p>
    </div>
);