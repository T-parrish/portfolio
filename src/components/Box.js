import React from 'react';

const Box = ({color, opacity}) => {
    const style = {
        display: 'inline-block',
        backgroundColor: color,
        opacity: opacity
    }
    
    return <div style={style} /> 
};

export default Box