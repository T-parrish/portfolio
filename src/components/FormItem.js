import React from 'react';

export const FormItem = ({content}) => (
    <div> 
        <li>{content}</li>
    </div>
);