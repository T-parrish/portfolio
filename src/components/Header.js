import React from 'react';
import { NavLink, Link } from 'react-router-dom';

export const Header = () => (
	<header className="header">
		<div className="content-container">
			<div className="nav__wrapper">
				<div className="nav-box__one fade--top">
					<h1 className="nav__title">Taylor Parrish</h1>
					<div className="nav__links">
						<NavLink to='/' activeClassName="is-active" exact={true}>Home </NavLink>
						<NavLink to='/portfolio' activeClassName="is-active">Portfolio </NavLink>
						<NavLink to='/boxy' activeClassName="is-active">Boxy </NavLink>
					</div>
				</div>
				<div className="nav-box__two fade--in">
					<a href="https://twitter.com/twpearish"><i className="fab fa-twitter"></i></a>
					<a href="https://www.instagram.com/et.and.me"><i className="fab fa-instagram"></i></a>
					<a href="www.linkedin.com/in/taylor-parrish-55663379"><i className="fab fa-linkedin-in"></i></a>
				</div>
			</div>
		</div>
	</header>
);

export default Header;