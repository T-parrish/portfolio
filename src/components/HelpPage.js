import React from 'react';

const HelpPage = () => (
	<div>
		<p>This is from my HelpPage component</p>
	</div>
);

export default HelpPage;