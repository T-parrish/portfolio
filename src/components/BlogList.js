import React from 'react';
import {BlogItem} from '../components/BlogItem';

// needs to map over BlogItems 
export class BlogList extends React.Component {
    constructor(props) {
        super(props)

    }

    render() {
        const blogs = this.props.blogs.map((blog, id) => (
            <BlogItem key={id} content={blog} />
        ));

        return (
            <div>
                {blogs}
            </div>
        )
    }

}