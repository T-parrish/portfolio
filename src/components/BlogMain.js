import React from 'react';
import {BlogItem} from '../components/BlogItem';

// BlogMain -> BlogItem -> BlogPage
// Need to request data from api with componentWillMount
// store that data in the state
// pass the data down

export class BlogMain extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            blogs: [{
                title: 'Deez',
                description: 'testing testing'
            }, {
                title: 'Doeze',
                description: 'testing testing'
            }, {
                title: 'DaEZe',
                description: 'testing testing'
            }]
        }
    }

    render() {
        const blogs = this.state.blogs.map((blog, id) => (
            <BlogItem key={id} content={blog} />
        ));

        return (
            <div className="content-container">
                {blogs}
            </div>
        )
    }
}