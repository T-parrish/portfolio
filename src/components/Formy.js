import React from 'react';
import { FormItem } from '../components/FormItem';
import { LoginForm } from '../components/LoginForm';

export class Formy extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            inputText:'',
            todos: []
        }
    
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleChange(e) {
        // sets the state based off of the input's name and value
        this.setState({[e.target.name]: e.target.value})
    }

    handleSubmit(e) {
        e.preventDefault()
        // object spreading returns a new array, tacks on the input text
        // this way you're not directly mutating the state.
        const todos = [...this.state.todos, this.state.inputText]
        this.setState({inputText:'', todos})
    }

    initLogin = (e) => {
        e.preventDefault();
        const creds = this.state
        console.log(this.props)
        this.props.startUserLogin(creds)
	};

    render () {
        // always remember to implicitly return the components 
        // with the map function... curly braces are a no-no.
        const todos = this.state.todos.map((todo, i) => (
            <FormItem key={i} content={todo} />
        ));

        return (
            <div className="content-container"> 
                <h1> Simple Todo App</h1>
                <LoginForm />
                <form className="form" onSubmit={this.handleSubmit}>
                    <input 
                        type='text' 
                        name='inputText'
                        autoComplete= 'off'
                        value={this.state.inputText} 
                        onChange={this.handleChange}
                    />
                    <button className="button_two" type='submit'>SUBMIT</button>
                </form>
                <div>
                    <ol>
                        {todos}
                    </ol>
                </div>
            </div>
        )
    }
}