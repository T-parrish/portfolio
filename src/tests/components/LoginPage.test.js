import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/LoginPage';

let startLogin, wrapper

beforeEach(() => {
    startLogin = jest.fn();
    wrapper = shallow(<LoginPage startLogin = {startLogin}/>);
});

test('should render Login Page correctly', () => {
    expect(wrapper).toMatchSnapshot();
    wrapper.find('button').simulate('click');
    expect(startLogin).toHaveBeenCalled();
});

