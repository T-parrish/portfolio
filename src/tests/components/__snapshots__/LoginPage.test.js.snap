// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Login Page correctly 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Boilerplate
    </h1>
    <p>
       Tagline for app 
    </p>
    <button
      className="button button--logon"
      onClick={[MockFunction]}
    >
       Login with Google 
    </button>
  </div>
</div>
`;
